libs/go/libp2p_node/aea/utils.go

```diff
@@ -0,0 +1,90 @@
+/* -*- coding: utf-8 -*-
+* ------------------------------------------------------------------------------
+*
+*   Copyright 2018-2019 Fetch.AI Limited
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*
+* ------------------------------------------------------------------------------
+ */
+
+package aea
+
+import (
+	"errors"
+
+	acn "libp2p_node/acn"
+	common "libp2p_node/common"
+
+	proto "google.golang.org/protobuf/proto"
+)
+
+func HandleAcnMessageFromPipe(
+	pipe common.Pipe,
+	statusQueue acn.StatusQueue,
+	counterpartyID string,
+) (*Envelope, error) {
+	envelope := &Envelope{}
+	var acn_err error
+
+	data, err := pipe.Read()
+
+	if err != nil {
+		logger.Error().Str("err", err.Error()).Msg("while receiving data")
+		return envelope, &common.PipeError{Err: err, Msg: "Pipe error during envelope read"}
+	}
+
+	msg_type, acn_envelope, status, err := acn.DecodeAcnMessage(data)
+
+	if err != nil {
+		logger.Error().Str("err", err.Error()).Msg("while decoding acn")
+		acn_err = acn.SendAcnError(pipe, "error on decoding acn message")
```

Comment:
maybe here we could be more precise on the type of error? e.g. `ERROR_SERIALIZATION` (see how it is used in other parts of the code)

Reply:
updated